gpu/octree/include/pcl/gpu/octree/octree.h

```diff
@@ -0,0 +1,213 @@
+
+            /** \brief Batch approximate nearest search on GPU
+              * \param[in] queries array of centers
+              * \param[out] result array of results ( one index for each query ) 
+              */
+            PCL_DEPRECATED(1, 14, "use approxNearestSearch() which returns square distances instead")
+            void approxNearestSearch(const Queries& queries, NeighborIndices& result) const;
+
+            /** \brief Batch approximate nearest search on GPU
+              * \param[in] queries array of centers
+              * \param[out] result array of results ( one index for each query )
+              * \param[out] sqr_distance corresponding square distances to results from query point
+              */
+            void approxNearestSearch(const Queries& queries, NeighborIndices& result, ResultSqrDists& sqr_distance) const;
+
+            /** \brief Batch exact k-nearest search on GPU for k == 1 only!
+              * \param[in] queries array of centers
+              * \param[in] k number of neighbors (only k == 1 is supported)
+              * \param[out] results array of results
+              */
+            void nearestKSearchBatch(const Queries& queries, int k, NeighborIndices& results) const;
+
+            /** \brief Batch exact k-nearest search on GPU for k == 1 only!
+              * \param[in] queries array of centers
+              * \param[in] k number of neighbors (only k == 1 is supported)
+              * \param[out] results array of results
+              * \param[out] sqr_distances square distances to results
+              */
+            void nearestKSearchBatch(const Queries& queries, int k, NeighborIndices& results, ResultSqrDists& sqr_distances) const;
+
+            /** \brief Desroys octree and release all resources */
+            void clear();            
+        private:
+            void *impl;            
+            bool built_;
+        };        
+
+      /** \brief Performs brute force radius search on GPU
+        * \param[in] cloud cloud where to search
+        * \param[in] query query point
+        * \param[in] radius radius
+        * \param[out] result indeces of points within give sphere
+        * \param[in] buffer buffer for intermediate results. Keep reference to it between calls to eliminate internal allocations
+        */             
+        template <class T>
+        PCL_EXPORTS void bruteForceRadiusSearchGPU(const typename Octree<T>::PointCloud& cloud, const typename Octree<T>::PointType& query, float radius, DeviceArray<int>& result, DeviceArray<int>& buffer);
+    }
+}
+}
+namespace pcl {
+namespace gpu {
+using Octree = pcl::gpu::details::Octree<pcl::PointXYZ>;
+
+const auto bruteForceRadiusSearchGPU =
+    pcl::gpu::details::bruteForceRadiusSearchGPU<pcl::PointXYZ>;
+
+template <class PointT>
+using OcTree [[deprectated("Will be replaced by Octree at PCL 1.15")]] =
```

Comment:
PCL_DEPRECATED does this, right?

Reply:
Good that you mention this! I will now finally learn about the PCL_DEPRECATED macro, and then update the PR.

Reply:
I tried to use the PCL macro. Do you agree with the way I used it?